src/test/java/seedu/address/logic/parser/EditCommandParserTest.java

```diff
@@ -1,203 +1,112 @@
 import static seedu.address.logic.parser.CliSyntax.PREFIX_GENRE;
 import static seedu.address.logic.parser.CommandParserTestUtil.assertParseFailure;
 import static seedu.address.logic.parser.CommandParserTestUtil.assertParseSuccess;
-import static seedu.address.testutil.TypicalIndexes.INDEX_FIRST_BOOK;
-import static seedu.address.testutil.TypicalIndexes.INDEX_SECOND_BOOK;
-import static seedu.address.testutil.TypicalIndexes.INDEX_THIRD_BOOK;
 
 import org.junit.jupiter.api.Test;
 
-import seedu.address.commons.core.index.Index;
-import seedu.address.logic.commands.EditCommand;
-import seedu.address.logic.commands.EditCommand.EditBookDescriptor;
-import seedu.address.model.book.SerialNumber;
-import seedu.address.model.genre.Genre;
-import seedu.address.testutil.EditBookDescriptorBuilder;
+import seedu.address.logic.commands.EditBorrowerCommand;
+import seedu.address.logic.commands.EditBorrowerCommand.EditBorrowerDescriptor;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.borrower.Email;
+import seedu.address.model.borrower.Phone;
+import seedu.address.testutil.EditBorrowerDescriptorBuilder;
 
 public class EditCommandParserTest {
 
     private static final String TAG_EMPTY = " " + PREFIX_GENRE;
 
     private static final String MESSAGE_INVALID_FORMAT =
-            String.format(MESSAGE_INVALID_COMMAND_FORMAT, EditCommand.MESSAGE_USAGE);
-
-    private static final String VALID_INDEX_ONE = "1";
+            String.format(MESSAGE_INVALID_COMMAND_FORMAT, EditBorrowerCommand.MESSAGE_USAGE);
 
     private EditCommandParser parser = new EditCommandParser();
 
     @Test
     public void parse_missingParts_failure() {
-        // no index specified
-        assertParseFailure(parser, VALID_TITLE_BOOK_1, MESSAGE_INVALID_FORMAT);
 
         // no field specified
-        assertParseFailure(parser, VALID_INDEX_ONE, EditCommand.MESSAGE_NOT_EDITED);
-
-        // no index and no field specified
-        assertParseFailure(parser, "", MESSAGE_INVALID_FORMAT);
+        assertParseFailure(parser, PREAMBLE_WHITESPACE, EditBorrowerCommand.MESSAGE_NOT_EDITED);
+        assertParseFailure(parser, "", EditBorrowerCommand.MESSAGE_NOT_EDITED);
     }
 
     @Test
-    public void parse_invalidPreamble_failure() {
-        // negative index
-        assertParseFailure(parser, "-5" + TITLE_DESC_BOOK_1, MESSAGE_INVALID_FORMAT);
-
-        // zero index
-        assertParseFailure(parser, "0" + TITLE_DESC_BOOK_1, MESSAGE_INVALID_FORMAT);
-
-        // invalid arguments being parsed as preamble
-        assertParseFailure(parser, VALID_INDEX_ONE + " some random string", MESSAGE_INVALID_FORMAT);
-
-        // invalid prefix being parsed as preamble
-        assertParseFailure(parser, VALID_INDEX_ONE + " i/ string", MESSAGE_INVALID_FORMAT);
+    public void parse_notInServeMode_failure() {
+        assertParseFailure(parser, PREAMBLE_WHITESPACE, EditBorrowerCommand.MESSAGE_NOT_EDITED);
```

Comment:
Does this check for serve mode / did we set `Model` to normal mode before testing this?

Reply:
the command itself checks for serve mode because one of the methods throws exception when it is not in serve mode.